drivers/scheduler/k8s/k8s.go

```diff
@@ -5113,11 +5134,89 @@ func (k *K8s) CreateCsiSnapsAndValidate(ctx *scheduler.Context, snapClass string
 	return nil
 }
 
-func (k *K8s) createSnapshotAndVerify(data, snapName, namespace, snapClass, restoredPVCName, originalPod, originalPVC string) error {
-	elasticsearchSC := "elasticsearch-sc"
+// CloneFAVolumeAndValidate create new PVC by cloning an existing PVC and make sure the contents of volume are same
+// (Testrail cases C58509)
+func (k *K8s) CloneFAVolumeAndValidate(ctx *scheduler.Context, request scheduler.CSICloneRequest)  error {
+	// This test will validate the content of the snapshot as opposed to just verify creation of snapshot.
+	timestamp := request.Timestamp
+	namespace := request.Namespace + timestamp
+	clientset, err := getKubeClient("")
+	if err != nil {
+		logrus.Errorf("Failed to get kube client for clone test: %s", err)
+		return err
+	}
+
+	storageClassName := request.StorageclassName
+	// for FADA test, find a FA SC on our own that has pure_block param, if no FA SC, skip the test
+	if request.FADA {
```

Comment:
If you really need this. Make this code snippet as a function. Prevent to repeat it twice.